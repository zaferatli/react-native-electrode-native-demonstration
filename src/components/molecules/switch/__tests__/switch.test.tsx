import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';

import Switch, {SwitchPropsType} from '../index';

const getComponent = (props: SwitchPropsType) => {
  return <Switch {...props} />;
};

it('initial render button', () => {
  renderer.create(getComponent({}));
});

it('Primary Switch', () => {
  const props: SwitchPropsType = {
    value: true,
  };
  const RenderedComponent = renderer.create(getComponent(props)).toJSON();
  expect(RenderedComponent).toMatchSnapshot();
});

it('Primary Switch With Color', () => {
  const props: SwitchPropsType = {
    trackColorFalse: 'red',
    trackColorTrue: 'green',
    thumbColorFalse: 'white',
    thumbColorTrue: 'white',
    value: true,
  };
  const RenderedComponent = renderer.create(getComponent(props)).toJSON();
  expect(RenderedComponent).toMatchSnapshot();
});

it('Primary Switch With Color False State', () => {
  const props: SwitchPropsType = {
    trackColorFalse: 'red',
    trackColorTrue: 'green',
    thumbColorFalse: 'white',
    thumbColorTrue: 'white',
    value: false,
  };
  const RenderedComponent = renderer.create(getComponent(props)).toJSON();
  expect(RenderedComponent).toMatchSnapshot();
});
