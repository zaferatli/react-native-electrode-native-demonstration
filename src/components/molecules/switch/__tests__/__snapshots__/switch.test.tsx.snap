// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary Switch 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "backgroundColor": undefined,
      "borderRadius": 14,
      "height": 31,
      "position": "relative",
      "width": 51,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": undefined,
          "borderRadius": 14,
          "height": 27,
          "left": 2,
          "position": "absolute",
          "top": 2,
          "width": 27,
        }
      }
    />
  </View>
</View>
`;

exports[`Primary Switch With Color 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "backgroundColor": "rgba(255, 0, 0, 1)",
      "borderRadius": 14,
      "height": 31,
      "position": "relative",
      "width": 51,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderRadius": 14,
          "height": 27,
          "left": 2,
          "position": "absolute",
          "top": 2,
          "width": 27,
        }
      }
    />
  </View>
</View>
`;

exports[`Primary Switch With Color False State 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "backgroundColor": "rgba(255, 0, 0, 1)",
      "borderRadius": 14,
      "height": 31,
      "position": "relative",
      "width": 51,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderRadius": 14,
          "height": 27,
          "left": 2,
          "position": "absolute",
          "top": 2,
          "width": 27,
        }
      }
    />
  </View>
</View>
`;
